### PR TITLE
Adding promise chains to sports spec

### PR DESCRIPTION
**Pull request by Sammyfrw**
@ShaiNirShado 

This'll be the PR for refactoring the controllers and spec files that I've done work on. I'll be commenting parts that I might want advice on, too.

### CODE REVIEW

File: server/components/populate_sport_test.js [hunk 1]
```diff
@@ -3,37 +3,40 @@
...
 var fs = require('fs')
+var BPromise = require("sequelize/node_modules/bluebird");
 
 var Populate = (function(){
   var _parseCsv = function(csv, sport) {
     var fileStream = fs.createReadStream(csv);
     var csvConverter = new Converter({constructResult:true});
-    csvConverter.on("end_parsed",function(jsonObj){
-      populateDatabase(jsonObj, sport);
+    return BPromise.try(function() {
```
Comment (Sammyfrw) [inline server/components/populate_sport_test.js]: I'm trying to fix the promise chain in this spec, which I think always gets broken when the CSV parser is run. I want to be able to actually have the promise chain incorporate the csv parser and wait until it finishes running so the spec works properly (in the sense that the chain isn't broken up at all so the first spec doesn't run until this is finished.)

Comment (ShaiNirShado) [inline server/components/populate_sport_test.js]: There's a way to take arbitrary functions and make them asynchronous with
promises. Take a look at promisification:
https://github.com/petkaantonov/bluebird/blob/master/API.md#promisification

There's an example in our code in message.helper.js in the sendEmail
function where I make e-mail sending return a promise so we can wait on
that.

File: server/api/sport/sport.controller.js [hunk 2]
```diff
@@ -72,9 +72,10 @@ exports.destroy = function(req, res) {
 exports.populate = function(req, res) {
     Sport.find(req.params.id).then(function (sport) {
         if(!sport) { return res.send(404); }
-        parseCsv(sport)
+        return parseCsv(sport)
+    }).then(function() {
         return res.send(201, sport);
-    }, function(error){
+    }.catch(function(err) {
```
Comment (Sammyfrw) [inline server/api/sport/sport.controller.js]: There's other segments in the controller that haven't swapped to using .catch. Should we change all of them to use this format, or would you rather keep it the way it is?

Comment (ShaiNirShado) [inline server/api/sport/sport.controller.js]: My tendency is to only refactor code I'm actively touching unless it's actually broken, so there's no need to go out of your way to switch it around.

**Comment by Sammyfrw** [comment]
What do you think about the branch so far? It's all basically refactoring. Good to merge?

File: server/api/sport/sport.controller.js [hunk 3]
```diff
@@ -72,9 +72,10 @@ exports.destroy = function(req, res) {
 exports.populate = function(req, res) {
     Sport.find(req.params.id).then(function (sport) {
         if(!sport) { return res.send(404); }
-        parseCsv(sport)
+        return parseCsv(sport)
+    }).then(function() {
         return res.send(201, sport);
-    }, function(error){
+    }).catch(function(err) {
```
Comment (ShaiNirShado) [inline server/api/sport/sport.controller.js]: Should be 'error' and not 'err' here

**Comment by ShaiNirShado** [comment]
Other than a tiny issue in error handling I noted, branch looks good to go.

**Comment by Sammyfrw** [comment]
Merged and closed.

**Pull request merged**